testing/environments/environments.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package environments
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/magefile/mage/mg"
+
+	"github.com/elastic/elastic-agent-shipper/tools"
+)
+
+type testImage struct {
+	name string
+	path string
+}
+
+type testImageConfig struct {
+	image       testImage         // The docker image
+	environment map[string]string // Environment variables for docker-compose
+}
+
+// The raw Docker images used by test configurations. New Docker images
+// should be added to this list.
+var (
+	elasticsearchImage = testImage{
+		name: "Elasticsearch",
+		path: "testing/environments/elasticsearch/docker-compose.yml",
+	}
+
+	kafkaImage = testImage{
+		name: "Kafka",
+		path: "testing/environments/kafka/docker-compose.yml",
+	}
+)
+
+// DefaultElasticsearch returns a configuration for an Elasticsearch server
+// with default settings listening on port 9200.
+func DefaultElasticsearch() TestImageConfig {
+	basePath := "docker.elastic.co/elasticsearch/elasticsearch"
+	version := tools.DefaultBeatVersion + "-SNAPSHOT"
+	return testImageConfig{
+		image: elasticsearchImage,
+		environment: map[string]string{
+			"ELASTICSEARCH_IMAGE_REF": fmt.Sprintf("%v:%v", basePath, version),
+		},
+	}
+}
+
+// DefaultKafka returns a configuration for a Kafka server with default
+// settings listening on port 9092.
+func DefaultKafka() TestImageConfig {
+	return testImageConfig{image: kafkaImage}
+}
+
+// Up calls docker-compose up on the given container configurations and returns the result.
+func Up(configs []TestImageConfig) error {
+	cmd := dockerComposeCommand([]string{"up", "-d"}, configs)
+
+	return cmd.Run()
```

Comment:
might be nice to use [Output instead of Run](https://pkg.go.dev/os/exec#Cmd.Output).  That way if there is an error we can show the stdout & stderr.  Otherwise you would have to re-run with `mg.Verbose` to see the error.